lib/elixir_sense/core/state.ex

```diff
@@ -286,12 +303,21 @@ defmodule ElixirSense.Core.State do
   def maybe_add_protocol_implementation(state, {protocol, implementations}) do
     implementation_modules =
       implementations
-      |> Enum.flat_map(fn module ->
-        expanded = expand_alias(state, Module.concat(module))
-        unescape_protocol_impementations(expanded)
+      |> Enum.flat_map(fn
+        module when is_list(module) ->
+          expanded = expand_alias(state, Module.concat(module))
+          unescape_protocol_impementations(expanded)
+
+        module when is_atom(module) ->
+          unescape_protocol_impementations(module)
       end)
 
-    candidate = expand_alias(state, Module.concat(protocol))
+    candidate =
+      if is_list(protocol) do
+        expand_alias(state, Module.concat(protocol))
```

Comment:
I feel like making `expand_alias` smarter with detection of `is_list` otherwise returning the thing itself may make some of this logic disappear.

Reply:
It could be done but the point of this logic is detecting if we are dealing with alias expression. If not then expanding it makes no sense.